### DISCUSSION TITLE
Does maturin support Python 'extra' dependencies (which are kind of like Rust's 'features')?

### DISCUSSION THREAD

**Discussion by CarlKCarlK** [PyO3/maturin#1835] [Q&A] [answered]
Rust has a notion of features. I think Python as something similar via 'extra' dependencies. Users might be able to say:

```
pip install bed-reader[samples]
```
to get the version of, for example, the [bed-reader](https://github.com/fastlmm/bed-reader/) genomics package that depends on Python's Pooch package to retrieve large sample files.

Does maturin support extra/features?

-- Carl
p.s. Sorry for posting multiple questions. We're looking at a major new update and we had a user decide not to use our package because they didn't want an unneeded-by-them dependency on Pooch.

**Comment by messense** [comment, marked as answer]
Use `optional-dependencies` as specified by PEP 621: https://peps.python.org/pep-0621/#dependencies-optional-dependencies

**Reply by CarlKCarlK** [reply]
Thanks! It's working great.

**Reply by jsalort** [reply]
Can you use `optional-dependencies` to specify an optional cargo feature? From the PEP, it seems to be used to specify optional Python dependency.
